isis/tests/BundleMeasureTests.cpp

```diff
@@ -0,0 +1,166 @@
+// }
+//
+//
+// TEST_F( BundleMeasure_CreateObj, CopyConstructor )
+// {
+//   BundleMeasure testBundleMeasure = *testBundleMeasurePtr;
+//
+//   // Using the isRejected Boolean to confirm that the testBundleMeasure
+//   // was copied correctly.
+//   // Note: The default value of isRejected is false.
+//   testBundleMeasure.setRejected( true );
+//
+//   // The copy constructor is then used on the testBundleMeasure object.
+//   BundleMeasure copiedBundleMeasure = BundleMeasure( testBundleMeasure );
+//
+//   // Confirm that this is a copy of the testBundleMeasure by checking the
+//   // isRejected value.
+//   EXPECT_TRUE( copiedBundleMeasure.isRejected() );
+//
+//   // Confirm that the copied BundleMeasure is not simply pointing to
+//   // the testBundleMeasure using the isRejected value again.
+//
+//   copiedBundleMeasure.setRejected( false );
+//
+//   EXPECT_FALSE( copiedBundleMeasure.isRejected() );
+//   EXPECT_TRUE( testBundleMeasure.isRejected() );
+// }
+//
+// TEST_F( BundleMeasure_CreateObj, CopyConstructor2 )
+// {
+//   // Using the isRejected Boolean to confirm that the testBundleMeasure
+//   // was copied correctly.
+//   // Note: The default value of isRejected is false.
+//   std::cout << "Setting Reject" << '\n';
+//   testBundleMeasurePtr->setRejected( true );
+//   std::cout << "Reject Set" << '\n';
+//
+//   // The copy constructor is then used on the testBundleMeasure object.
+//   BundleMeasure copiedBundleMeasure( *testBundleMeasurePtr );
+//
+//   // Confirm that this is a copy of the testBundleMeasure by checking the
+//   // isRejected value.
+//   EXPECT_TRUE( copiedBundleMeasure.isRejected() );
+//
+//   // Confirm that the copied BundleMeasure is not simply pointing to
+//   // the testBundleMeasure using the isRejected value again.
+//
+//   copiedBundleMeasure.setRejected( false );
+//
+//   EXPECT_FALSE( copiedBundleMeasure.isRejected() );
+//   EXPECT_TRUE( testBundleMeasurePtr->isRejected() );
+// }
+//
+// TEST_F( BundleMeasure_CreateObj, AssignmentOperator )
+// {
+//   // Using the isRejected Boolean to confirm that the testBundleMeasure
+//   // was assigned correctly.
+//   // Note: The default value of isRejected is false.
+//   //testBundleMeasurePtr->setRejected( true );
+//   BundleMeasure assignedBundleMeasure = *testBundleMeasurePtr;
+//
+//   // Using the isRejected Boolean to confirm that the testBundleMeasure
+//   // was assigned correctly.
+//   // Note: The default value of isRejected is false.
+//   //assignedBundleMeasure.setRejected( true );
+//
+//
+//
+//
+// }
+//
+// TEST_F( BundleMeasure_CreateObj, IsRejected )
+// {
+//   EXPECT_EQ( false, testBundleMeasurePtr->isRejected() );
+//   testBundleMeasurePtr->setRejected( true );
+//   EXPECT_EQ( true, testBundleMeasurePtr->isRejected() );
+//   testBundleMeasurePtr->setRejected( false );
+//   EXPECT_EQ( false, testBundleMeasurePtr->isRejected() );
+//
+// }
```

Comment:
Any reason most of the tests are commented out?

Reply:
Mostly to focus on the issue we think is causing a lot of the problems with the rest if the tests. For instance, the first CopyConstructor test doesn't cause errors, but fails on the final assertion, and CopyConstructor2 causes a segfault to occur, despite doing essentially the same thing as the first CopyConstructor.  
Also, the first two lines of code in the AssignmentOperator test cause a segfault, but running the test with only one line or the other will not.